x-pack/packages/kbn-elastic-assistant/impl/knowledge_base/knowledge_base_settings_management/document_entry_editor.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiCheckbox,
+  EuiFieldText,
+  EuiForm,
+  EuiFormRow,
+  EuiMarkdownEditor,
+  EuiSuperSelect,
+  EuiIcon,
+  EuiText,
+} from '@elastic/eui';
+import React, { useCallback } from 'react';
+import { DocumentEntry } from '@kbn/elastic-assistant-common';
+import * as i18n from './translations';
+
+interface Props {
+  entry?: DocumentEntry;
+  setEntry: React.Dispatch<React.SetStateAction<Partial<DocumentEntry>>>;
+}
+
+export const DocumentEntryEditor: React.FC<Props> = React.memo(({ entry, setEntry }) => {
+  // Name
+  const setName = useCallback(
+    (e) => setEntry((prevEntry) => ({ ...prevEntry, name: e.target.value })),
+    [setEntry]
+  );
+
+  // Sharing
+  const setSharingOptions = useCallback(
+    (value) =>
+      setEntry((prevEntry) => ({
+        ...prevEntry,
+        users: value === i18n.SHARING_GLOBAL_OPTION_LABEL ? [] : undefined,
```

Comment:
Consider eliminating the `Sharing` dropdown in the screenshot below, because the `Global to everyone in the Space` option doesn't appear to be applicable anymore:

![sharing](https://github.com/user-attachments/assets/20c14135-b497-4886-b80d-e8d54de5d6c1)

Reply:
That makes sense. I was thinking we shouldn't provide this option without RBAC, because it wouldn't be possible to create global entries, but it's the opposite: effectively everyone's an admin.